src/System.IO.Pipelines/PipelineConnectionStream.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    public class PipelineConnectionStream : Stream
```

Comment:
>Stream [](start = 44, length = 6)

I think the adapters should be separated (namespacewise) from the pure-pipeline APIs